packages/resource-deployment/scripts/function-app-create.sh

```diff
@@ -109,72 +109,44 @@ publishFunctionAppScripts() {
     echo "  Successfully fetched principal ID $principalId."
 }
 
-deployWebApiArmTemplate() {
-    functionAppNamePrefix="web-api-allyfuncapp"
-    templateFilePath="${0%/*}/../templates/function-web-api-app-template.json"
+deployFunctionApp() {
+    local functionAppNamePrefix=$1
+    local templateFilePath=$2
+    local functionAppName=$3
+    local extraParameters=$4
 
-    echo "Deploying Azure Function App using ARM template..."
+    echo "Deploying Azure Function App $functionAppName using ARM template..."
     resources=$(az group deployment create \
         --resource-group "$resourceGroupName" \
         --template-file "$templateFilePath" \
-        --parameters clientId="$webApiAdClientId" namePrefix="$functionAppNamePrefix" releaseVersion="$releaseVersion" \
+        --parameters namePrefix="$functionAppNamePrefix" releaseVersion="$releaseVersion" $extraParameters \
         --query "properties.outputResources[].id" \
         -o tsv)
 
     . "${0%/*}/get-resource-name-from-resource-paths.sh" -p "Microsoft.Web/sites" -r "$resources"
-    webApiFunctionAppName="$resourceName"
+    local myFunctionAppName="$resourceName"
 
-    waitForFunctionAppServiceDeploymentCompletion $webApiFunctionAppName
-    echo "Successfully deployed Azure Function App '$webApiFunctionAppName'"
-}
-
-deployWebWorkersArmTemplate() {
-    functionAppNamePrefix="web-workers-allyfuncapp"
-    templateFilePath="${0%/*}/../templates/function-web-workers-app-template.json"
-
-    echo "Deploying Azure Function App using ARM template..."
-    resources=$(az group deployment create \
-        --resource-group "$resourceGroupName" \
-        --template-file "$templateFilePath" \
-        --parameters namePrefix="$functionAppNamePrefix" releaseVersion="$releaseVersion" \
-        --query "properties.outputResources[].id" \
-        -o tsv)
-
-    . "${0%/*}/get-resource-name-from-resource-paths.sh" -p "Microsoft.Web/sites" -r "$resources"
-    webWorkersFunctionAppName="$resourceName"
-
-    waitForFunctionAppServiceDeploymentCompletion $webWorkersFunctionAppName
-    echo "Successfully deployed Azure Function App '$webWorkersFunctionAppName'"
-}
+    waitForFunctionAppServiceDeploymentCompletion $myFunctionAppName
+    echo "Successfully deployed Azure Function App '$myFunctionAppName'"
 
-deployWebApiFunctionApp() {
-    deployWebApiArmTemplate $webApiPackageName
-
-    # Keep child script call only one function level deep to preserve exports
-    getFunctionAppPrincipalId $webApiFunctionAppName
+    getFunctionAppPrincipalId $myFunctionAppName
     . "${0%/*}/key-vault-enable-msi.sh"
-}
 
-deployWebWorkersFunctionApp() {
-    deployWebWorkersArmTemplate $webWorkersPackageName
-
-    # Keep child script call only one level deep to preserve exports
-    getFunctionAppPrincipalId $webWorkersFunctionAppName
-    . "${0%/*}/key-vault-enable-msi.sh"
+    eval $functionAppName="'$myFunctionAppName'"
```

Comment:
why not just functionAppName="'$myFunctionAppName'" ?

Reply:
also why is this line needed? who is using it?

Reply:
I don't want to set functionAppName, I want to set it's value, this is one way to handle Returning Values from Bash Functions
so the command (eval $functionAppName="'$myFunctionAppName'") will be compiled as 
(eval webAPiFunctionAppName="'$myFunctionAppName'") for webApi call and as
(eval webWorkerFunctionAppName="'$myFunctionAppName'") for webWorker call

Look at this: https://www.linuxjournal.com/content/return-values-bash-functions

Reply:
P2: i see. you could use get-resource-names.sh to get the names.